packages/dashboard/src/components/alert-store.tsx

```diff
@@ -17,90 +19,49 @@ export interface AlertToDisplay extends RobotWithTask {
   show: boolean;
 }
 
-const returnOnlyRobotsWithTask = (value: Record<string, RobotWithTask[]>) => {
-  return Object.values(value)
-    .flatMap((r) => r)
-    .filter((element) => {
-      if (element.task) {
-        return true;
-      }
-      return false;
-    });
-};
-
 export const AlertStore = React.memo(() => {
   const rmf = React.useContext(RmfAppContext);
-
-  const [fleets, setFleets] = React.useState<string[]>([]);
-  const [robotsWithTasks, setRobotsWithTasks] = React.useState<Record<string, RobotWithTask[]>>({});
+  const [taskAlerts, setTaskAlerts] = React.useState<Record<string, Alert>>({});
+  const [robotAlerts, setRobotAlerts] = React.useState<Record<string, Alert>>({});
 
   React.useEffect(() => {
     if (!rmf) {
       return;
     }
-
-    const sub = rmf.fleetsObs.subscribe((fleets) => {
-      setFleets(
-        fleets.reduce<string[]>((acc, f) => {
-          if (f.name) {
-            acc.push(f.name);
-          }
-          return acc;
-        }, []),
-      );
+    const sub = rmf.alertObsStore.subscribe(async (alert) => {
+      if (alert.category === 'task') {
+        setTaskAlerts((prev) => {
+          return {
+            ...prev,
+            [alert.id]: alert,
+          };
+        });
+      } else if (alert.category === 'fleet' || alert.category === 'robot') {
+        setRobotAlerts((prev) => {
+          return {
+            ...prev,
+            [alert.id]: alert,
+          };
+        });
+      }
```

Comment:
We can use the Switch statement here to be more clear. 
```suggestion
      switch (alert.category) {
        case 'task':
          setTaskAlerts((prev) => ({ ...prev, [alert.id]: alert }));
          break;
        case 'fleet':
        case 'robot':
          setRobotAlerts((prev) => ({ ...prev, [alert.id]: alert }));
          break;
        default:
        // default case
      }

Reply:
Good idea, this is much cleaner. https://github.com/open-rmf/rmf-web/pull/694/commits/7fddd62cdf8e9e2e69122a2005adcc8543a23f4e